Update order of build steps in README

If attempting to run `mvn compile` before `git submodule init && git submodule update` the following error results:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11.465 s
[INFO] Finished at: 2021-05-21T15:27:50-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.11.3:yarn (yarn run build) on project cryostat: Failed to run task: 'yarn run build' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
```

Following this, `web-client` is non-empty which prevents the `git submodule init && git submodule update` commands from succeeding. We should edit the README to mention the submodule steps before the `mvn compile` step.

--- a/README.md
+++ b/README.md
@@ -39,10 +39,10 @@ required dependency, which is not currently published in an artefact repository
 and so must be built and installed into the Maven local repository.
 Instructions for doing so are available at that project's README.
 
+Submodules must be initialized via `git submodule init && git submodule update`.
+
 Once the `cryostat-core` local dependency is made available,
 `mvn compile` will build the project.
-
-Submodules must be initialized via `git submodule init && git submodule update`.
 
 Unit tests can be run with `mvn test`. Integration tests and additional quality
 tools can be run with `mvn verify`.

